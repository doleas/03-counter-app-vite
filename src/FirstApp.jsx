import PropTypes from 'prop-types';

//React encontro la solucion para no tener que importar Fragment o tener que ponerlo dentro de un contenedor div
// import { Fragment } from "react";

// const FirstApp = () => {
//     return (
//         <Fragment>
//             <h3 style={{ textAlign: "center" }}>FirstApp</h3>
//             <p style={{ textAlign: "center" }}> React JS</p>
//         </Fragment>
//     )
// }

// const FirstApp = () => {
//     return (
//         //esto tiene <> el mismo comportamiento que un fragmento o un nodo padre
//         <> 
//             <h3 style={{ textAlign: "center" }}>FirstApp</h3>
//             <p style={{ textAlign: "center" }}> React JS</p>
//         </>
//     )
// }

//esto es permitido
const mensaje = "Master en Ingeniería de Software";

//imprimir todo el objeto es permitido solo si se llama cada variable o si se imprime tipo JSON
const obj = {
    message: 'Bienvenido',
    title: 'App'
}

const edad = 42;
//creamos una funcion
const SecondApp = (a, b) => {
    return a * b;

}


// const FirstApp = () => {

//     console.log();
//     return (
//         //esto tiene <> el mismo comportamiento que un fragmento o un nodo padre
//         <> 
//             {/* <h3 style={{ textAlign: "center" }}>{obj.message}</h3> podemos traer la informacion de una variable*/}
//             {/* <div style={{ textAlign: "center" }}><code>{JSON.stringify(SecondApp())}</code> </div> */}
//             <div style={{ textAlign: "center" }}><h3>Tengo {SecondApp(6, 7)} años de edad</h3> </div>
//             <h3 style={{ textAlign: "center" }}>{obj.message}</h3> 
//             <p style={{ textAlign: "center" }}> React JS</p>
//         </>
//     )
// }


const FirstApp = ({title, subtitle, name}) => {

    // if (!title){
    //     throw new Error('El titulo no existe!!');
    // }



    return (
        <>  
            <h1 style={{textAlign: "center"}}>{title}</h1>
            <h3 style={{textAlign: "center"}}>{subtitle}</h3>
            <p style={{textAlign: "center"}}>Soy un titulo !!</p>
            <p style={{textAlign: "right"}}>{name}</p>
        </>
    )
}


export default FirstApp;

FirstApp.propTypes = {

    name: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired, //es requerido
}

//defaultprops van a entrar antes que nuestros proptypes
FirstApp.defaultProps = {
    name: 'David Oleas',
    subtitle: 983818763,
    // title: 'Innovación Tecnológica'
}
