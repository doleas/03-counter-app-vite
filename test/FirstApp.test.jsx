import { render } from "@testing-library/react"
import FirstApp from "../src/FirstApp"


describe('pruebas en <FirstApp />', () => {

    test('Debe hacer match con el snapshot', () => {

        const title = 'Hola, soy el Sr Goku';
        const { container } = render(<FirstApp title={title} />);

        expect(container).toMatchSnapshot();

    });

    test('Debe mostrar el titulo en un h1', () => {

        const title = 'Hola, soy el Sr Goku';
        const { container, getByText } = render(<FirstApp title={title} />);

        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');
        // console.log(h1.innerHTML);
        // expect(h1.innerHTML).toBe(title);
        expect(h1.innerHTML).toContain(title);

    });

    test('Debe mostrar el subtitulo eviado por props', () => {

        const title = 'Hola, soy el Sr Goku';
        const subtitle = 'Hola soy un subtitulo';
        const { getAllByText } = render(
            <FirstApp
                title={title}
                subtitle={subtitle}
            />);

        expect(getAllByText(subtitle).length ).toBe(1);
       
    });

});