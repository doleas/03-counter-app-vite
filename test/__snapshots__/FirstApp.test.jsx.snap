// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    style="text-align: center;"
  >
    Hola, soy el Sr Goku
  </h1>
  <h3
    style="text-align: center;"
  >
    Hola soy un subtitulo
  </h3>
  <p
    style="text-align: center;"
  >
    Soy un titulo !!
  </p>
  <p
    style="text-align: right;"
  >
    David Oleas
  </p>
</div>
`;
